### PR TITLE
feat: case submission satisfaction survey(FPLA-1925)

### PR DESCRIPTION
**Pull request by lewisbirks**
### JIRA link (if applicable) ###

[FPLA-1925](https://tools.hmcts.net/jira/browse/FPLA-1925)

### Change description ###

Introduce a new page at the end of the case submission.
Page design with markdown and variables defined by the defaults provided by [StringSubstitutor](https://commons.apache.org/proper/commons-text/apidocs/org/apache/commons/text/StringSubstitutor.html).
Markdown separated into 2 chuncks through the use of an invalid html tag `<break>`.

Updated integration tests to now return `SubmittedCallbackResponse` for `postSubmittedEvent`.
Altered e2e tests to take into account the possibility of the confirmation page.

Flux PR https://github.com/hmcts/cnp-flux-config/pull/4519

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/markdown/CaseSubmissionMarkdownServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Test
+    void shouldSplitOnSeparator() {
+        MarkdownData markdownData = service.getMarkdownData(CASE_NAME);
+
+        assertThat(markdownData).isEqualTo(buildExpectedData());
+    }
+
+    @Test
+    void shouldRemoveInitialBlankLinesWhenCaseNameNotProvided() {
```
Comment (Jamesnelson117) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/markdown/CaseSubmissionMarkdownServiceTest.java:24]: minor: shouldTrimBlankLinesWhenCaseNameNotProvided

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/markdown/CaseSubmissionMarkdownServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        MarkdownData markdownData = service.getMarkdownData(null);
+
+        assertThat(markdownData).isEqualTo(buildExpectedData(""));
+    }
+
+    private MarkdownData buildExpectedData() {
```
Comment (Jamesnelson117) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/markdown/CaseSubmissionMarkdownServiceTest.java:30]: minor: buildExpectedMarkdownData

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/markdown/CaseSubmissionMarkdownServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        return buildExpectedData(CASE_NAME);
+    }
+
+    private MarkdownData buildExpectedData(String caseName) {
```
Comment (Jamesnelson117) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/markdown/CaseSubmissionMarkdownServiceTest.java:34]: minor: buildMarkdownData

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/markdown/MarkdownSubstitutionServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+import uk.gov.hmcts.reform.fpl.model.markdown.CaseSubmissionSubstitutionData;
+import uk.gov.hmcts.reform.fpl.model.markdown.MarkdownData;
+import uk.gov.hmcts.reform.fpl.model.markdown.MarkdownSubstitutionData;
+
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.fpl.enums.MarkdownTemplate.CASE_SUBMISSION;
+
+class MarkdownSubstitutionServiceTest {
+    private static final String CASE_NAME = "Gio vs FPLA";
```
Comment (Jamesnelson117) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/markdown/MarkdownSubstitutionServiceTest.java:14]: :O

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/markdown/MarkdownSubstitutionService.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+    protected abstract MarkdownData transform(String templateData);
+
+    private String substitute(String templateData, MarkdownSubstitutionData markdownData) {
+        return replace(templateData, mapper.<Map<String, String>>convertValue(markdownData, new TypeReference<>() {}));
+    }
+
+    private String readMarkdownTemplate(MarkdownTemplate template) {
+        return ResourceReader.readString(String.format("%s/%s.md", TEMPLATE_DIRECTORY, template.getFile()));
+    }
```
[the remark this answers is not in the document]
Comment (lewisbirks) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/markdown/MarkdownSubstitutionService.java:35]: Don't think so, if there is a problem with the file we should throw an error which is done in the `ResourceReader` and propagate it to the top. This should mean that they just don't see the screen and a purple banner instead.

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/markdown/CaseSubmissionMarkdownServiceTest.java (hunk 3 above)
Comment (lewisbirks) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/markdown/CaseSubmissionMarkdownServiceTest.java:34]: `buildExpectedMarkdownData` instead

**Pull request merged**